pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,291 @@
+			Zone:        opt.Zone,
+		}
+
+		_, err = gcpclient.Disks().Insert(opt.ProjectID, opt.Zone, disk).Context(ctx).Do()
+		if err != nil {
+			return nil, fmt.Errorf("%w, failed to create disk", err)
+		}
+	}
+
+	logger.Info("Disk created", "disk", opt.DiskName)
+
+	disks := []*compute.AttachedDisk{
+		{
+			AutoDelete: true,
+			Boot:       true,
+			DiskSizeGb: 10,
+			Source:     "projects/" + opt.ProjectID + "/zones/" + opt.Zone + "/disks/" + opt.DiskName,
+			Mode:       "READ_WRITE",
+		},
+	}
+
+	networkInterfaces := []*compute.NetworkInterface{
+		{
+			Network:       opt.Network,
+			Subnetwork:    opt.Subnetwork,
+			AccessConfigs: []*compute.AccessConfig{{Name: "External NAT", Type: "ONE_TO_ONE_NAT"}},
+		},
+	}
+
+	machineType := "zones/" + opt.Zone + "/machineTypes/n1-standard-1"
+
+	instance, err := getBastionInstance(ctx, gcpclient, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	metadataItems := []*compute.MetadataItems{
+		{
+			Key:   "startup-script",
+			Value: pointer.StringPtr(string(bastion.Spec.UserData)),
+		},
+	}
+
+	if instance != nil {
+		logger.Info("Existing bastion compute instance found", "compute_instance_name", instance.Name)
+	} else {
+		instance := &compute.Instance{
+			Disks:              disks,
+			DeletionProtection: false,
+			Description:        "Bastion Instance",
+			Name:               opt.BastionInstanceName,
+			Zone:               opt.Zone,
+			MachineType:        machineType,
+			NetworkInterfaces:  networkInterfaces,
+			Tags:               &compute.Tags{Items: []string{opt.BastionInstanceName}},
+			Metadata:           &compute.Metadata{Items: metadataItems},
+		}
+
+		_, err = gcpclient.Instances().Insert(opt.ProjectID, opt.Zone, instance).Context(ctx).Do()
+		if err != nil {
+			return nil, fmt.Errorf("%w, failed to create instance", err)
+		}
+	}
+
+	return getInstanceEndpoints(ctx, gcpclient, opt)
+}
+
+func getInstanceEndpoints(ctx context.Context, gcpclient gcpclient.Interface, opt *Options) (*bastionEndpoints, error) {
+	instance, err := getBastionInstance(ctx, gcpclient, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	if instance == nil {
+		return nil, nil
+	}
+
+	if instance.Status != "RUNNING" {
+		return nil, fmt.Errorf("Instance not RUNNING, Status:" + instance.Status)
+
+	}
+
+	endpoints := &bastionEndpoints{}
+
+	networkInterfaces := instance.NetworkInterfaces
+	internalIP := &networkInterfaces[0].NetworkIP
+	externalIP := &networkInterfaces[0].AccessConfigs[0].NatIP
+
+	if len(networkInterfaces) == 0 {
+		return nil, fmt.Errorf(instance.Name + ":" + "no network interfaces found")
+	}
+
+	if len(networkInterfaces[0].AccessConfigs) == 0 {
+		return nil, fmt.Errorf(instance.Name + ":" + "no access config found for network interface")
+	}
+
+	if ingress := addressToIngress(&instance.Name, internalIP); ingress != nil {
+		endpoints.private = ingress
+	}
+
+	// public dns name not supported now
+	// https://cloud.google.com/compute/docs/instances/create-ptr-record#api
```

Comment:
```suggestion
	// gcp does not automatically assign a public dns name to the instance (in contrast to e.g. aws). As we provide an externalIP to connect to the bastion, having a public dns name would just be an alternative way to connect to the bastion. Out of this reason, we spare the effort to create a PTR record (see https://cloud.google.com/compute/docs/instances/create-ptr-record#api) just for the sake of having it.
```
this sounds as if it

Reply:
committed in fbad5cfe22865b73a53ff0684b28bd8563af8e5d